From 5356f8ee65ca2735ff690412f85f403c97c1b6c3 Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Wed, 5 Aug 2020 00:25:22 +0300
Subject: [PATCH 78/79] imx8m-mini: u-boot: Add __weak
 uboot_board_private_init()

Add a __weak uboot_board_private_init().
Each board can redefind this function in its board.c file.

Signed-off-by: Valentin Raevsky <valentin@compulab.co.il>
---
 board/compulab/plat/imx8mm/board/board.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/board/compulab/plat/imx8mm/board/board.c b/board/compulab/plat/imx8mm/board/board.c
index cf0b8eb0ff..a753042d91 100644
--- a/board/compulab/plat/imx8mm/board/board.c
+++ b/board/compulab/plat/imx8mm/board/board.c
@@ -7,6 +7,7 @@
 #include <common.h>
 #include <malloc.h>
 #include <errno.h>
+#include <hang.h>
 #include <asm/io.h>
 #include <miiphy.h>
 #include <netdev.h>
@@ -181,12 +182,21 @@ int board_usb_cleanup(int index, enum usb_init_type init)
 
 	return 0;
 }
+
+__weak int uboot_board_private_init(void) {
+	return 0;
+}
+
 int board_init(void)
 {
 
 #ifdef CONFIG_FEC_MXC
 	setup_fec();
 #endif
+	if (uboot_board_private_init()) {
+		printf("uboot_board_private_init() failed\n");
+		hang();
+	}
 	return 0;
 }
 
-- 
2.11.0

